allennlp/semparse/contexts/table_question_context.py

```diff
@@ -115,25 +201,24 @@ def get_table_knowledge_graph(self) -> KnowledgeGraph:
             # now, and later add number and string entities as needed.
             number_columns = []
             date_columns = []
-            for column, column_type in self.column_types.items():
-                if column_type == "number":
-                    column_name = f"number_column:{column}"
-                    number_columns.append(column_name)
-                elif column_type == "date":
-                    column_name = f"date_column:{column}"
-                    date_columns.append(column_name)
-                else:
-                    column_name = f"string_column:{column}"
+            for typed_column_name in self.table_data[0].keys():
+                if "number_column:" in typed_column_name or "num2_column" in typed_column_name:
+                    number_columns.append(typed_column_name)
+
+                if "date_column:" in typed_column_name:
+                    date_columns.append(typed_column_name)
+
                 # Add column names to entities, with no neighbors yet.
-                entities.add(column_name)
-                neighbors[column_name] = []
-                entity_text[column_name] = column.replace("_", " ")
+                entities.add(typed_column_name)
+                neighbors[typed_column_name] = []
+                entity_text[typed_column_name] = typed_column_name.split(":")[-1].replace("_", " ")
```

Comment:
It's probably worth doing `typed_column_name.split(":", 1)[-1]`, so that you don't accidentally truncate the text.

Reply:
Thanks!